source/time.tex

```diff
@@ -8039,713 +8077,340 @@
+\tcode{duration<int, ratio<1, 6>>} & \tcode{6} & \tcode{0.166666}    \\ \rowsep
+\tcode{duration<int, ratio<1, 7>>} & \tcode{6} & \tcode{0.142857}    \\ \rowsep
+\tcode{duration<int, ratio<1, 8>>} & \tcode{3} & \tcode{0.125}       \\ \rowsep
+\tcode{duration<int, ratio<1, 9>>} & \tcode{6} & \tcode{0.111111}    \\ \rowsep
+\tcode{duration<int, ratio<1, 10>>} & \tcode{1} & \tcode{0.1}        \\ \rowsep
+\tcode{duration<int, ratio<756, 625>>} & \tcode{4} & \tcode{0.2096}  \\
+\end{LongTable}
+\end{example}
 \end{itemdescr}
 
-\indexlibrarymember{make24}{time_of_day<hours>}%
 \begin{itemdecl}
-constexpr void make24() noexcept;
+using precision = @\seebelow@;
 \end{itemdecl}
 
 \begin{itemdescr}
 \pnum
-\effects
-If \tcode{*this} is a 12-hour time,
-converts to a 24-hour time.
-Otherwise, no effects.
+\tcode{precision} is
+\begin{codeblock}
+duration<common_type_t<Duration::rep, seconds::rep>, ratio<1, @$10^\tcode{fractional_width}$@>>
+\end{codeblock}
 \end{itemdescr}
 
-\indexlibrarymember{make12}{time_of_day<hours>}%
 \begin{itemdecl}
-constexpr void make12() noexcept;
+constexpr explicit hh_mm_ss(Duration d);
 \end{itemdecl}
 
 \begin{itemdescr}
 \pnum
-\effects
-If \tcode{*this} is a 24-hour time,
-converts to a 12-hour time.
-Otherwise, no effects.
-\end{itemdescr}
-
-\rSec2[time.tod.minutes]{Minutes precision}
-\indexlibrary{\idxcode{time_of_day<minutes>}}
-
-\begin{codeblock}
-namespace std::chrono {
-  template<>
-  class time_of_day<minutes> {
-  public:
-    using precision = chrono::minutes;
-
-    time_of_day() = default;
-    constexpr explicit time_of_day(chrono::minutes since_midnight) noexcept;
-
-    constexpr chrono::hours    hours()   const noexcept;
-    constexpr chrono::minutes  minutes() const noexcept;
-
-    constexpr explicit operator precision()    const noexcept;
-    constexpr          precision to_duration() const noexcept;
-
-    constexpr void make24() noexcept;
-    constexpr void make12() noexcept;
-  };
-}
-\end{codeblock}
-
-\pnum
+\effect
+Constructs an object of type \tcode{hh_mm_ss}
+which represents the \tcode{Duration d} with precision \tcode{precision}.
+\begin{itemize}
+\item
+  Initializes \tcode{is_neg} with \tcode{d < Duration::zero()}.
+\item
+  Initializes \tcode{h} with \tcode{duration_cast<chrono::hours>(abs(d))}.
+\item
+  Initializes \tcode{m}
+  with \tcode{duration_cast<chrono::minutes>(abs(d) - hours())}.
+\item
+  Initializes \tcode{s}
+  with \tcode{duration_cast<chrono::seconds>(abs(d) - hours() - minutes())}.
+\item
+  If \tcode{treat_as_floating_point_v<precision::rep>} is \tcode{true},
+  initializes \tcode{ss} with \tcode{abs(d) - hours() - minutes() - seconds()}.
+  Otherwise, initializes \tcode{ss}
+  with \tcode{duration_cast<precision>(abs(d) - hours() - minutes() - seconds())}.
+\end{itemize}
 \begin{note}
-This specialization handles hours and minutes since midnight.
+When \tcode{precision} is \tcode{seconds} with integral representation,
```

Comment:
I think this note shouldn't use `seconds` in code font, since its referring to the concept of seconds rather than the specific implementation of `seconds` which already has an integral representation. I suggest replacing "is `seconds`" with "represents seconds", or to be more precise (perhaps unnecessarily for a note), "has a tick period of a second".

Reply:
"precision" is a typedef, and "seconds" is some chrono type, so a C++-level entity. So this part seems correct. Maybe the "integral representation" is not a restriction, but falls out from the rest, and should thus be phrased differently?

Reply:
Considering the nature of notes, it seems better to just do away with "with integral representation". The statement would still hold true when `precision` is `duration<int128_t>`, which also represents "`seconds`", but is not `seconds`.

Reply:
> Maybe the "integral representation" is not a restriction, but falls out from the rest, and should thus be phrased differently?

The statement is only true when `precision::rep` is an integral type, not for floating-point types. But it's also true for types other than `chrono::seconds`, e.g. `chrono::minutes` and `chrono::hours` and `chrono::duration<short, ratio<60>>`.

> Considering the nature of notes, it seems better to just do away with "with integral representation". The statement would still hold true when precision is `duration<int128_t>`, which also represents "seconds", but is not `seconds`.

But it would not hold true for `duration<double>` which represents seconds, but can have subsecond precision.

A more accurate (but less clear) wording would be "when `treat_as_floating_point_v<precision::rep>` is `false` and `is_same_v<precision::period, ratio<1>>` is `true` ...".

@HowardHinnant ?

Reply:
Or instead of using `is_same`, use `ratio_equal_v<precision::period, ration<1>>`.

But arguably the verbosity makes the note less useful.

Reply:
Jonathan has a complete understanding of this issue.

@johelegp 's suggestion above including "has a tick period of a second" seemed pretty on-the-money to me.  The _non-normative_  note applies when `rep` is integral and `period` is `ratio<1>`.

Reply:
Also, technically speaking, `0s` isn't always returned, but instead `precision{0}`, the difference being that `0s` has type `std::chrono::seconds`.

I don't have strong feelings about how this is resolved.  The note doesn't specify anything that isn't specified elsewhere and is meant to be informative on a point that otherwise might be confusing.  Thus the more relaxed language.  The less confusing the better, whatever the editors deem less confusing.

Reply:
Here's a shot at it:

```
\begin{note}
When \tcode{precision::rep} is integral and \tcode{precision::period} is \tcode{ratio<1>},
\tcode{subseconds()} always returns a value equal to \tcode{0s}.
\end{note}
```

Reply:
Fixed.